conan/cli/commands/compare.py

```diff
@@ -0,0 +1,233 @@
+import json
+import subprocess
+import os
+
+from conan.errors import ConanException
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+from conan.internal.util.files import chdir
+from conan.api.model import RecipeReference, ListPattern
+
+def output_json(msg):
+    return json.dumps({"greet": msg})
+
+
+@conan_command(group="Security", formatters={"json": output_json})
+def compare(conan_api: ConanAPI, parser, *args):
+    """
+    Command to get the diff between versions
+    """
+
+    parser.add_argument("-op", "--old-path", help="Path to the old recipe")
+    parser.add_argument("-ov", "--old-reference", help='Old reference "mylib/1.0"')
+    parser.add_argument("-or", "--old-require", help='Old reference "mylib/1.0"')
```

Comment:
What is the difference? Maybe this can be simplified to just 1?

Reply:
Done, good idea, now the flow is:
* Defining _just_ `--old-reference` will fetch that reference from the remotes
* But if `--old-path` is set, `--old-reference` will be used to set the reference for the conanfile that is being pointed at

The only downside is that if the reference is set in the recipe itself, you'll have to match it on the cmd, but we can work something out if it turns out to be too much of a pain